### PR TITLE
Feature/add javadoc and log4j

### PR DESCRIPTION
**Pull request by jbarbaso**
Added Javadocs to each project file.
Created logger interceptor.
Added logger interceptor annotation to the car resource, car service and entities.
Added logger methods to each exception.
Created Postman test collection.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,5 +60,15 @@
 		<artifactId>swagger-jaxrs</artifactId>
 		<version>1.5.22</version>
 	</dependency>
+	<dependency>
+	    <groupId>org.glassfish.jersey.core</groupId>
```
Comment (javieraviles) [inline pom.xml:64]: what's this dependency for?

File: src/main/java/com/everis/cars/boundary/CarResource.java [hunk 2]
```diff
@@ -32,6 +41,13 @@
 	@Produces(MediaType.APPLICATION_JSON)
 	public List<Car> getAllCars();
 
+	/**
+	 * Create and retrieve the created car
+	 * 
+	 * @param car the car object to be created
```
Comment (javieraviles) [inline src/main/java/com/everis/cars/boundary/CarResource.java]: you can use @link to reference an object or class in javadocs

File: pom.xml (hunk 1 above)
Comment (jbarbaso) [reply pom.xml:64]: This dependency was added to be used in PathParamExceptionMapper.
It's to send a formated exception message for org.glassfish.jersey.server.ParamException.PathParamException class.

File: src/main/java/com/everis/cars/boundary/CarResource.java [hunk 3]
```diff
@@ -53,6 +69,15 @@ public Response createCar(
 		final Car car
 	);
 
+	/**
+	 * Update a car by ID
+	 * 
+	 * @param carId the car ID to be updated
+	 * @param car the car object to update the registry
+	 * @throws CarNotFoundException
```
Comment (javieraviles) [inline src/main/java/com/everis/cars/boundary/CarResource.java]: is it really throwing the exception? worth maybe mentioning error codes here

Comment (javieraviles) [reply src/main/java/com/everis/cars/boundary/CarResource.java]: even mentioning how BeanValConstrainViolationExceptionMapper will intercept returning a bad request?

File: src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
 
+	/**
+	 * Logger instance
+	 * 
+	 * @see org.apache.log4j.Logger
+	 */
+	private static Logger logger = Logger.getLogger(BeanValConstrainViolationExceptionMapper.class);
+	
+	/**
+	 * toResponse override to implement the custom response message
```
Comment (javieraviles) [inline src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java]: what the main purpose of this?

Comment (javieraviles) [reply src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java]: rest layer to throw 400 when...

File: src/main/java/com/everis/cars/exceptions/CarNotFoundExceptionMapper.java [hunk 5]
```diff
@@ -5,14 +5,34 @@
...
+	/**
+	 * Logger instance
+	 * 
+	 * @see org.apache.log4j.Logger
+	 */
+	private static Logger logger = Logger.getLogger(CarNotFoundException.class);
 
+	/**
+	 * toResponse override to implement the custom response message 
+	 *
```
Comment (javieraviles) [inline src/main/java/com/everis/cars/exceptions/CarNotFoundExceptionMapper.java:31]: purpose of this? 404 in resources when...

**Review by javieraviles — CHANGES_REQUESTED** [review]
review some comments please

File: src/main/java/com/everis/cars/boundary/CarResource.java (hunk 2 above)
Comment (jbarbaso) [reply src/main/java/com/everis/cars/boundary/CarResource.java]: I added the @link annotation to the comments to point the class commentend in javadocs.

File: src/main/java/com/everis/cars/boundary/CarResource.java (hunk 3 above)
Comment (jbarbaso) [reply src/main/java/com/everis/cars/boundary/CarResource.java]: > is it really throwing the exception? worth maybe mentioning error codes here

explained better where and why this exception is throwing.



> even mentioning how BeanValConstrainViolationExceptionMapper will intercept returning a bad request?

explained in comments how I catch the bean validations and the return sended

File: src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java (hunk 4 above)
Comment (jbarbaso) [reply src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java]: I explained better this methods in exception mappers.

File: src/main/java/com/everis/cars/exceptions/CarNotFoundExceptionMapper.java (hunk 5 above)
Comment (jbarbaso) [reply src/main/java/com/everis/cars/exceptions/CarNotFoundExceptionMapper.java:31]: I explained better this methods in exception mappers.

**Comment by jbarbaso** [comment]
> review some comments please

Comments reviewed

**Review by javieraviles — APPROVED** [review]

**Pull request merged**